perl/man/man3/oose.3
.\" Automatically generated by Pod::Man 2.28 (Pod::Simple 3.31)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "oose 3"
.TH oose 3 "2015-08-27" "perl v5.22.1" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
oose \- syntactic sugar to make Moose one\-liners easier
.SH "VERSION"
.IX Header "VERSION"
version 2.1603
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.Vb 2
\&  # create a Moose class on the fly ...
\&  perl \-Moose=Foo \-e \*(Aqhas bar => ( is=>q[ro], default => q[baz] ); print Foo\->new\->bar\*(Aq # prints baz
\&
\&  # loads an existing class (Moose or non\-Moose)
\&  # and re\-"opens" the package definition to make
\&  # debugging/introspection easier
\&  perl \-Moose=+My::Class \-e \*(Aqprint join ", " => _\|_PACKAGE_\|_\->meta\->get_method_list\*(Aq
\&
\&  # also loads Moose::Util::TypeConstraints to allow subtypes etc
\&  perl \-Moose=Person \-e\*(Aqsubtype q[ValidAge] => as q[Int] => where { $_ > 0 && $_ < 78 }; has => age ( isa => q[ValidAge], is => q[ro]); Person\->new(age => 90)\*(Aq
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
oose.pm is a simple source filter that adds
\&\f(CW\*(C`package $name; use Moose; use Moose::Util::TypeConstraints;\*(C'\fR
to the beginning of your script and was entirely created because typing
\&\f(CW\*(C`perl \-e\*(Aqpackage Foo; use Moose; ...\*(Aq\*(C'\fR was annoying me.
.SH "INTERFACE"
.IX Header "INTERFACE"
oose provides exactly one method and it's automatically called by perl:
.IP "\fBimport($package)\fR" 4
.IX Item "import($package)"
Pass a package name to import to be used by the source filter. The
package defaults to \f(CW\*(C`Class\*(C'\fR if none is given.
.SH "DEPENDENCIES"
.IX Header "DEPENDENCIES"
You will need Filter::Simple and eventually Moose
.SH "INCOMPATIBILITIES"
.IX Header "INCOMPATIBILITIES"
None reported. But it is a source filter and might have issues there.
.SH "BUGS"
.IX Header "BUGS"
See \*(L"\s-1BUGS\*(R"\s0 in Moose for details on reporting bugs.
.SH "AUTHORS"
.IX Header "AUTHORS"
.IP "\(bu" 4
Stevan Little <stevan.little@iinteractive.com>
.IP "\(bu" 4
Dave Rolsky <autarch@urth.org>
.IP "\(bu" 4
Jesse Luehrs <doy@tozt.net>
.IP "\(bu" 4
Shawn M Moore <code@sartak.org>
.IP "\(bu" 4
יובל קוג'מן (Yuval Kogman) <nothingmuch@woobling.org>
.IP "\(bu" 4
Karen Etheridge <ether@cpan.org>
.IP "\(bu" 4
Florian Ragwitz <rafl@debian.org>
.IP "\(bu" 4
Hans Dieter Pearcey <hdp@weftsoar.net>
.IP "\(bu" 4
Chris Prather <chris@prather.org>
.IP "\(bu" 4
Matt S Trout <mst@shadowcat.co.uk>
.SH "COPYRIGHT AND LICENSE"
.IX Header "COPYRIGHT AND LICENSE"
This software is copyright (c) 2006 by Infinity Interactive, Inc..
.PP
This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


CHANGES.txt
ActivePerl 5.22 Change Log
    For the latest information on ActivePerl, please see:

        http://www.ActiveState.com/ActivePerl/

  Build 2201 January, 2016
    Build 2201 is based on Perl 5.22.1 plus additional selected changes.

    *Bug Fixes and Changes since build 2200*

    *  The following security vulnerabilities have been addressed:

       OpenSSL - multiple fixes
           OpenSSL has been upgraded to 1.0.2e to address several moderate
           issues. Please see
           https://www.openssl.org/news/secadv/20151203.txt for full
           details.

           CVE-2015-3193: BN_mod_exp may produce incorrect results on
           x86_64

           CVE-2015-3194: Certificate verify crash with missing PSS
           parameter

           CVE-2015-3195: X509_ATTRIBUTE memory leak

           CVE-2015-3196: Race condition handling PSK identify hint

           CVE-2015-1794: Anon DH ServerKeyExchange with 0 p parameter

       CVE-2015-8607
           Beginning in PathTools 3.47 and/or perl 5.20.0, the
           File::Spec::canonpath() routine returned untained strings even
           if passed tainted input. This defect undermines the guarantee
           of taint propagation, which is sometimes used to ensure that
           unvalidated user input does not reach sensitive code.

       CVE-2015-8608
           Perl 5 on Windows suffers from two out-of-bounds read and
           multiple small buffer over-read vulnerabilities in the
           VDir::MapPathA and VDir::MapPathW functions that could
           potentially be exploited to achieve arbitrary code execution.

           These defects have been present since
           perl-5.005_02-2346-g7766f13, circa 1999, and are still present
           in the latest releases of Perl.

    *  Significant changes that have occurred in the Perl 5.22.1 release
       are documented in perl5221delta.

    *  Most bundled modules have been updated to their latest released
       version from CPAN. Use the "ppm query" command to check the exact
       version included in this release.

  Build 2200 June, 2015
    *What's new in the 2200 Series*

    *  This build corresponds to the Perl 5.22.0 source code release.

    *  The 2200 series builds of ActivePerl are not binary compatible with
       builds in the 2000 or earlier series. Any extensions built using
       binaries from the ActivePerl 2000 or earlier series will need to be
       recompiled. Note especially that this applies to PPM packages that
       may have been built for earlier series of ActivePerl.

    *  Significant changes that have occurred since the 5.20 release are
       documented in perl5220delta.

    *  Most bundled distributions have been updated to their latest
       released version from CPAN. Use the "ppm list" command to check the
       exact version included in this release.



perl/lib/oose.pm
package oose;
our $VERSION = '2.1603';

use strict;
use warnings;

use Moose::Util ();

BEGIN {
    my $package;
    sub import {
        $package = $_[1] || 'Class';
        if ($package =~ /^\+/) {
            $package =~ s/^\+//;
            Moose::Util::_load_user_class($package);
        }
    }
    use Filter::Simple sub { s/^/package $package;\nuse Moose;use Moose::Util::TypeConstraints;\n/; }
}

1;

# ABSTRACT: syntactic sugar to make Moose one-liners easier

__END__

=pod

=encoding UTF-8

=head1 NAME

oose - syntactic sugar to make Moose one-liners easier

=head1 VERSION

version 2.1603

=head1 SYNOPSIS

  # create a Moose class on the fly ...
  perl -Moose=Foo -e 'has bar => ( is=>q[ro], default => q[baz] ); print Foo->new->bar' # prints baz

  # loads an existing class (Moose or non-Moose)
  # and re-"opens" the package definition to make
  # debugging/introspection easier
  perl -Moose=+My::Class -e 'print join ", " => __PACKAGE__->meta->get_method_list'

  # also loads Moose::Util::TypeConstraints to allow subtypes etc
  perl -Moose=Person -e'subtype q[ValidAge] => as q[Int] => where { $_ > 0 && $_ < 78 }; has => age ( isa => q[ValidAge], is => q[ro]); Person->new(age => 90)'

=head1 DESCRIPTION

oose.pm is a simple source filter that adds
C<package $name; use Moose; use Moose::Util::TypeConstraints;>
to the beginning of your script and was entirely created because typing
C<perl -e'package Foo; use Moose; ...'> was annoying me.

=head1 INTERFACE

oose provides exactly one method and it's automatically called by perl:

=over 4

=item B<import($package)>

Pass a package name to import to be used by the source filter. The
package defaults to C<Class> if none is given.

=back

=head1 DEPENDENCIES

You will need L<Filter::Simple> and eventually L<Moose>

=head1 INCOMPATIBILITIES

None reported. But it is a source filter and might have issues there.

=head1 BUGS

See L<Moose/BUGS> for details on reporting bugs.

=head1 AUTHORS

=over 4

=item *

Stevan Little <stevan.little@iinteractive.com>

=item *

Dave Rolsky <autarch@urth.org>

=item *

Jesse Luehrs <doy@tozt.net>

=item *

Shawn M Moore <code@sartak.org>

=item *

יובל קוג'מן (Yuval Kogman) <nothingmuch@woobling.org>

=item *

Karen Etheridge <ether@cpan.org>

=item *

Florian Ragwitz <rafl@debian.org>

=item *

Hans Dieter Pearcey <hdp@weftsoar.net>

=item *

Chris Prather <chris@prather.org>

=item *

Matt S Trout <mst@shadowcat.co.uk>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2006 by Infinity Interactive, Inc..

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut
